modules/eks/loki/main.tf

```diff
@@ -0,0 +1,190 @@
+
+  verify          = var.verify
+  wait            = var.wait
+  atomic          = var.atomic
+  cleanup_on_fail = var.cleanup_on_fail
+  timeout         = var.timeout
+
+  eks_cluster_oidc_issuer_url = replace(module.eks.outputs.eks_cluster_identity_oidc_issuer, "https://", "")
+
+  iam_role_enabled = true
+  iam_policy = [{
+    statements = [
+      {
+        sid    = "AllowLokiStorageAccess"
+        effect = "Allow"
+        resources = [
+          module.loki_storage["chunks"].bucket_arn,
+          module.loki_storage["ruler"].bucket_arn,
+          module.loki_storage["admin"].bucket_arn,
+          format("%s/*", module.loki_storage["chunks"].bucket_arn),
+          format("%s/*", module.loki_storage["ruler"].bucket_arn),
+          format("%s/*", module.loki_storage["admin"].bucket_arn),
+        ]
+        actions = [
+          "s3:ListBucket",
+          "s3:PutObject",
+          "s3:GetObject",
+          "s3:DeleteObject"
+        ]
+      },
+    ]
+  }]
+
+  values = compact([
+    yamlencode({
+      loki = {
+        # For new installations, schema config doesnt change. See the following:
+        # https://grafana.com/docs/loki/latest/operations/storage/schema/#new-loki-installs
+        schemaConfig = {
+          configs = [
+            {
+              from         = "2024-04-01" # for a new install, this must be a date in the past, use a recent date. Format is YYYY-MM-DD.
+              object_store = "s3"
+              store        = "tsdb"
+              schema       = "v13"
+              index = {
+                prefix = "index_"
+                period = "24h"
+              }
+            }
+          ]
+        }
+        storage = {
+          bucketNames = {
+            chunks = module.loki_storage["chunks"].bucket_id
+            ruler  = module.loki_storage["ruler"].bucket_id
+            admin  = module.loki_storage["admin"].bucket_id
+          },
+          type = "s3",
+          s3 = {
+            region = var.region
+          }
+        }
+      }
+      # Do not use the default nginx gateway
+      gateway = {
+        enabled = false
+      }
+      # Instead, we want to use AWS ALB Ingress Controller
+      ingress = {
+        enabled = true
+        annotations = {
+          "kubernetes.io/ingress.class"               = "alb"
+          "external-dns.alpha.kubernetes.io/hostname" = local.ingress_host_name
+          "alb.ingress.kubernetes.io/group.name"      = local.ingress_group_name
+          # We dont need to supply "alb.ingress.kubernetes.io/certificate-arn" because of AWS ALB controller's auto discovery using the given host
+          "alb.ingress.kubernetes.io/backend-protocol" = "HTTP"
+          "alb.ingress.kubernetes.io/listen-ports"     = "[{\"HTTP\": 80},{\"HTTPS\":443}]"
+          "alb.ingress.kubernetes.io/ssl-redirect"     = "443"
+          "alb.ingress.kubernetes.io/scheme"           = "internal"
+          "alb.ingress.kubernetes.io/target-type"      = "ip"
```

Comment:
this should probably just be defaulted by our yaml file

Reply:
Fine to leave to another PR

Reply:
I actually disagree. I prefer complexity in the component rather than stack configuration. Plus this way we can dynamically pull hostname and group name like on line 149,150. 

Although we can leave that conversation to another PR if we decide otherwise